src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelHelpers.cs

```diff
@@ -1672,20 +80,12 @@ private static Exception CreateUnexpectedResponseException(HttpRequestException
 
             return TraceResponseException(
                 new ProtocolException(SR.Format(SR.UnexpectedHttpResponseCode,
-                (int)response.StatusCode, statusDescription), responseException));
-        }
-
-        public static Exception CreateNullReferenceResponseException(NullReferenceException nullReferenceException)
-        {
-            return TraceResponseException(
-                new ProtocolException(SR.NullReferenceOnHttpResponse, nullReferenceException));
+                (int)response.StatusCode, statusDescription)));
```

Comment:
For background on why this was needed, `HttpWebRequest` would throw an exception for some HTTP status codes which means we would have an exception to set as the inner exception for the `ProtocolException`. With `HttpClient`, you get an `HttpResponseMessage` with the relevant status code and no exception. This means a lot of the failure modes for the Http transport no longer occur with an exception and instead are detected by inspecting the response code.